Add missing Blur trades

hildobby opened the pull request:
Brief comments on the purpose of your changes:

Add Blur trades routed directly through Seaport.

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

jeff-dude commented on models/blur/ethereum/blur_ethereum_events.sql:
```diff
@@ -134,4 +134,74 @@ LEFT JOIN {{ ref('nft_ethereum_transfers') }} seller_fix ON seller_fix.block_tim
+, tr.token_standard AS token_standard
+, nft_tok.name AS collection
+, CASE WHEN get_json_object(s.offer[0], '$.amount')=1 THEN 'Single Item Trade' ELSE 'Bundle Trade' END AS trade_type
+, get_json_object(s.offer[0], '$.amount') AS number_of_items
```
minor point, but can you cast this column (and all others that do so in above query) to the same datatypes? while spark will infer correctly based on above query, it's more for consistency/readability. this will apply to a few columns i'm sure, using this one as example to comment

jeff-dude commented on models/blur/ethereum/blur_ethereum_events.sql:
```diff
@@ -134,4 +134,74 @@ LEFT JOIN {{ ref('nft_ethereum_transfers') }} seller_fix ON seller_fix.block_tim
+FROM {{ source('seaport_ethereum','Seaport_evt_OrderFulfilled') }} s
+INNER JOIN {{ source('ethereum', 'transactions') }} tx ON tx.block_number=s.evt_block_number
+    AND tx.hash=s.evt_tx_hash
+    {% if is_incremental() %}
```
please add if not incremental logic in this join on the base table using project start date, similar to above query

jeff-dude commented on models/blur/ethereum/blur_ethereum_events.sql:
```diff
@@ -134,4 +134,74 @@ LEFT JOIN {{ ref('nft_ethereum_transfers') }} seller_fix ON seller_fix.block_tim
+    {% if is_incremental() %}
+    AND tx.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+LEFT JOIN tokens_ethereum.nft nft_tok ON nft_tok.contract_address=get_json_object(s.offer[0], '$.token')
```
convert table to jinja sytnax

jeff-dude commented on models/blur/ethereum/blur_ethereum_events.sql:
```diff
@@ -134,4 +134,74 @@ LEFT JOIN {{ ref('nft_ethereum_transfers') }} seller_fix ON seller_fix.block_tim
+    AND tx.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+LEFT JOIN tokens_ethereum.nft nft_tok ON nft_tok.contract_address=get_json_object(s.offer[0], '$.token')
+LEFT JOIN tokens_ethereum.erc20 currency_tok ON currency_tok.contract_address=get_json_object(s.consideration[0], '$.token')
```
convert table to jinja sytnax

jeff-dude commented on models/blur/ethereum/blur_ethereum_events.sql:
```diff
@@ -134,4 +134,74 @@ LEFT JOIN {{ ref('nft_ethereum_transfers') }} seller_fix ON seller_fix.block_tim
+LEFT JOIN {{ ref('prices_usd_forward_fill') }} pu ON ((pu.contract_address=get_json_object(s.consideration[0], '$.token') AND pu.blockchain='ethereum')
+        OR (get_json_object(s.consideration[0], '$.token')='0x0000000000000000000000000000000000000000'  AND pu.blockchain IS NULL AND pu.contract_address IS NULL AND pu.symbol='ETH'))
+    AND pu.minute=date_trunc('minute', s.evt_block_time)
+    {% if is_incremental() %}
```
please add if not incremental logic in this join on the prices table using project start date, similar to above query

jeff-dude commented on models/blur/ethereum/blur_ethereum_events.sql:
```diff
@@ -134,4 +134,74 @@ LEFT JOIN {{ ref('nft_ethereum_transfers') }} seller_fix ON seller_fix.block_tim
+    AND tr.contract_address=get_json_object(s.offer[0], '$.token')
+    AND tr.token_id=get_json_object(s.offer[0], '$.identifier')
+    AND tr.from=s.offerer
+    {% if is_incremental() %}
```
please add if not incremental logic in this join on the transfers table using project start date, similar to above query

jeff-dude commented on models/blur/ethereum/blur_ethereum_events.sql:
```diff
@@ -134,4 +134,74 @@ LEFT JOIN {{ ref('nft_ethereum_transfers') }} seller_fix ON seller_fix.block_tim
+    {% if is_incremental() %}
+    AND tr.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+WHERE s.evt_block_time >= '2023-01-25'
```
rather than hardcode this here, for best practice with jinja syntax, can we move this to not incremental block below (or via an else block)?

pandajackson42 commented:
Great work as always! Just curious, how about use another `version` value for these trades, to differentiate them from trades executed by the previous system? @hildobby

hildobby commented:
> Great work as always! Just curious, how about use another `version` value for these trades, to differentiate them from trades executed by the previous system? @hildobby

I think the goal of version is to match the version number as envisioned by the team behind the project and this is still Blur  v1 afaik (Blur is yet to even fully launch). If you're looking to filter for those, `project_contract_address` will be `0x000000000000ad05ccc4f10045630fb830b95127` when routed through Blur's direct marketplace contract and `0x00000000006c3852cbef3e08e8df289169ede581` when routed through seaport (all the new trades added in this PR).

hildobby commented:
Addresses all the points you mentioned @jeff-dude, seems ready to me now. If you merge, just an FYI that I extended the CI runtime to 120 minutes so that needs to be reverted first!
